### PR TITLE
Editor sizing

### PR DESCRIPTION
**Pull request by masweeney**
This fixes issue #1697, so that the editor gadget now respects the min/max-row/col parameters when calculating its preferred size.

P.S. new to `git` - let me know if I screwed this up

### CODE REVIEW

**Comment by AlexIljin** [comment]
I'll comment on the git log messages. The accepted template is:

```
<vocab(s)>: <short-summary>

[long-summary]
```

All lines should not exceed 75 characters. For example: `ui.gadgets.editors: editor pref-dim* uses max/min-row/column preferences`

Another note: no need to include the merge commit in the pull request. It's up to the receiving repository to decide whether to pull as merge or as rebase.

Finally, all these three commits could be squashed together, if there is no value in them being separate.

File: basis/ui/gadgets/editors/editors.factor [hunk 1]
```diff
@@ -213,9 +213,37 @@ M: editor draw-gadget*
         [ draw-lines ] [ draw-caret ] bi
     ] with-variable ;
 
+<PRIVATE
+
+: row,col-dim ( font r c -- dim )
+    "" resize-string [ drop CHAR: x ] map
```
Comment (mrjbq7) [inline basis/ui/gadgets/editors/editors.factor]: This should just be `CHAR: x <string>`.

File: basis/ui/gadgets/editors/editors.factor [hunk 2]
```diff
@@ -213,9 +213,37 @@ M: editor draw-gadget*
...
+    swap "" resize-string [ drop CHAR: \n ] map 
+    append text-dim ;
+
+: min-dim ( font editor -- dim )
+    [ min-rows>> 0 [ and ] most ]
```
Comment (mrjbq7) [inline basis/ui/gadgets/editors/editors.factor]: Could do `[ 0 ] unless`, I think.

File: basis/ui/gadgets/editors/editors.factor [hunk 3]
```diff
@@ -213,9 +213,37 @@ M: editor draw-gadget*
...
+    
+PRIVATE>
+
 M: editor pref-dim*
     ! Add some space for the caret.
-    [ font>> ] [ control-value ] bi text-dim { 1 0 } v+ ;
+    ! Now respects min/max-row/col parameters
+    [ font>> ] keep
+    [ max-dim ]
+    [ txt-dim ]
```
Comment (mrjbq7) [inline basis/ui/gadgets/editors/editors.factor]: I realize you are lining up with min and max but we try not to do too much abbreviation.

File: basis/ui/gadgets/editors/editors.factor (hunk 2 above)
Comment (AlexIljin) [inline basis/ui/gadgets/editors/editors.factor]: Yes, but with a star: `[ 0 ] unless*`.

**Comment by masweeney** [comment]
Thank you for the feedback. I submitted a different pull request (#1699) with these changes, in addition to those suggested in issue #1697.

**Pull request closed without merging**